docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
-# Developer Guide
+# CCA Manager Developer Guide
 
-## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+- [1. Introduction](#1-introduction)
+- [2. Setting up](#2-setting-up)
+- [3. Design and Implementation](#3-design-and-implementation)
+  * [3.1. Input Parsing](#31-input-parsing)
+  * [3.2. Commands](#32-commands)
+  * [3.3. Finance](#33-finance)
+  * [3.4. Event](#34-event)
+  * [3.5. HR](#35-hr)
+  * [3.6. Storage](#36-storage)
+- [4. Product scope](#4-product-scope)
+  * [4.1. Target user profile](#41-target-user-profile)
+  * [4.2. Value proposition](#42-value-proposition)
+- [5. User Stories](#5-user-stories)
+- [6. Non-Functional Requirements](#6-non-functional-requirements)
+- [7. Glossary](#7-glossary)
+- [8. Instructions for manual testing](#8-instructions-for-manual-testing)
 
 
-## Product scope
-### Target user profile
+## 1. Introduction 
+CCA Manager is a revolutionary all-in-one management tool that changes the way you can manage interest groups with unrivaled efficiency and simplicity. Its lightweight Command Line Interface (CLI) allows administrators to breeze through tasks quickly and easily while offering powerful features to advanced users.
 
-{Describe the target user profile}
+This developer guide is written to document the implementation of CCA Manager. This document is intended for people who
+are interested in learning more about the technical details of the various features and the organization of the application.
 
-### Value proposition
+## 2. Setting up
+Refer to the guide here.
 
-{Describe the value proposition: what problem does it solve?}
+## 3. Design and Implementation
 
-## User Stories
+![Architecture](BackendDiagram/Architecture.png)
+
+The **Architecture Design** given above explains the high-level design of the App. Given below is a quick overview of each component.
+
+**`Duke`** is the main class of the application, and handles the app launch, initializing the appropriate classes to be used.
+
+The rest of the app consists of the below:
+
+* [**`UI`**] : The UI of the App.
```

Comment:
As a newcomer to the application, I may not be familiar with the different components of the architecture. Perhaps a more detailed description with regards to each component can be helpful in explaining more clearly the function of each particular component!

For instance,

* [**`UI`**] : The UI of the App. Handles all console outputs onto the CLI interface.
* [**`Logic`**] : The command executor. <Describe what is a command and what it means to execute the command.>